app/components/rest/clusters.ts

```diff
@@ -1,11 +1,16 @@
 /// <reference path="../../../typings/tsd.d.ts" />
 
+import {ServerService} from '../rest/server'
+
 export class ClusterService {
 	config: Array<any>;
-	rest: restangular.ICollection;
-	static $inject: Array<string> = ['Restangular'];
-	constructor(rest:restangular.ICollection) {
-		this.rest = rest;
+    private clusterId: any;
+    private clusterModel: any;
+	static $inject: Array<string> = ['Restangular', '$location', '$q', 'ServerService'];
+	constructor(private rest:restangular.ICollection, 
+         private locationSvc: ng.ILocationService,
+         private iqSvc: ng.IQService,
```

Comment:
just rename these to $location and $q. So we can easily differentiate builtin services